### DISCUSSION TITLE
Why are my headings magenta?

### DISCUSSION THREAD

**Discussion by eamonburns** [vscode-neovim/vscode-neovim#1730] [Q&A]
This happens in HTML and Markdown (possibly other languages too). And it goes away when I disable the VSCode NeoVim plugin.  (ignore the squiggly lines, it's a markdown linter)

<img width="225" alt="image" src="https://github.com/vscode-neovim/vscode-neovim/assets/116777388/db108b5d-6635-44a7-84cb-b141edc335f9">

There are also some very strange visual bugs when I try to edit the magenta text

<img width="189" alt="image" src="https://github.com/vscode-neovim/vscode-neovim/assets/116777388/1f7757b6-07a2-401b-9426-d6e20a819f34">

(I placed the cursor in the text [the cursor disappeared] and inserted a space. But then an extra exclamation point appeared at the end instead of inserting a space in the right place)

This is what it looks like after reloading (no other edits)
<img width="186" alt="image" src="https://github.com/vscode-neovim/vscode-neovim/assets/116777388/9a30d331-bf44-4dff-9bbe-d011b23efd71">

**Comment by theol0403** [comment]
Does this happen with an empty init.vim?

**Reply by xiyaowong** [reply]
It was caused by syntax. Some highlights were not cleared as expected. I have encountered the same problem before in Markdown.

**Reply by ian-h-chamberlain** [reply]
This might be related to #1739 — I noticed that I most easily reproduced it when `init.vim` took a nontrivial amount of time but sometimes could still reproduce even with clean startup. 

As mentioned below it can be worked around with autocomands / disable `highlight` for some specific cases but I think it can theoretically occur with any file type that nvim provides highlights for.

**Reply by eamonburns** [reply]
@ian-h-chamberlain I think you're right. A few days ago, the highlighting for a bunch of other things (other than just headings, and not only magenta) went kind of crazy. So I looked through the extension's settings and found the `--clean` option, and it fixed the new problems, and I haven't seen the magenta headings after that either

**Comment by xiyaowong** [comment]
Please check the output of `:echo w:_vscode_hl_ns`

**Reply by eamonburns** [reply]
The output was `2`

**Reply by eamonburns** [reply]
Do you know how I might be able to fix it?

**Reply by xiyaowong** [reply]
You need to find out which vim highlight group it is.

**Reply by xiyaowong** [reply]
try `:hi clear Title`

**Reply by eamonburns** [reply]
`:hi clear Title` worked. Will I have to do that every time?

**Reply by xiyaowong** [reply]
In theory, this highlight should not appear and I am unable to reproduce it as well. For the time being, you can add an autocmd to clear it.
